### PR TITLE
DUOS-1315[risk=no]Consent: Delete deprecated methods from DataAccessRequestResource and Service

### PR DESCRIPTION
**Pull request by raejohanek**
SCOPE:
- delete deprecated API: describeDraftDar
- delete unused service methods
- refactor usages of service methods, rm usages from electionService and summaryService
- - refactor electionService and summaryService tests 
- no tests existed for the deleted API and service methods
- this endpoint "/partial/{id}" does not have documentation for a get method so nothing to delete in api-docs

Must be merged after other DUOS-1315 PRs so that it can compile. This PR deletes methods used by several deprecated APIs that could not be deleted in PRs deleting only one at a time.

ADDRESSES:
partially https://broadworkbench.atlassian.net/browse/DUOS-1315

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by JVThomas — COMMENTED** [review]
Nothing wrong with this PR, however the new user update endpoint does not update rationale and vote on a researcher review (see [DUOS-1314](https://github.com/DataBiosphere/duos-ui/pull/1145) for details). Once the fixes have been made this PR should be good to go.

File: src/main/java/org/broadinstitute/consent/http/resources/DACUserResource.java [hunk 1]
```diff
@@ -111,7 +111,6 @@ public Response update(@Auth AuthUser authUser, @Context UriInfo info, String js
         }
     }
 
-    @Deprecated // Use update instead
     @PUT
```
Comment (raejohanek) [inline src/main/java/org/broadinstitute/consent/http/resources/DACUserResource.java:106]: we no longer want to deprecate this API

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java [hunk 2]
```diff
@@ -478,7 +407,7 @@ public DataAccessRequest cancelDataAccessRequest(String referenceId) {
         List<DataSet> dataSetsToApprove = dataSetDAO.
                 findNeedsApprovalDataSetByDataSetId(datasetIdsForDatasetsToApprove);
 
-        // Sort documents by sort time, create time, then reversed.
+        // Sort dars by sort time, create time, then reversed.
         Comparator<DataAccessRequest> sortField = Comparator.comparing(d -> d.getData().getSortDate());
         Comparator<DataAccessRequest> createField = Comparator.comparing(d -> d.getData().getCreateDate());
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java:400]: Not for this PR, but we should be sorting by the dates on the dar, not the dar data.

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java:400]: And it should be update date, then create date, then reversed.

File: src/main/java/org/broadinstitute/consent/http/resources/DataRequestCasesResource.java [hunk 3]
```diff
@@ -35,22 +35,6 @@ public DataRequestCasesResource(ElectionService electionService, PendingCaseServ
         this.summaryService = summaryService;
     }
 
-    @GET
-    @Path("/pending")
-    @RolesAllowed({CHAIRPERSON, MEMBER})
-    public Response getDataRequestPendingCasesByAuthUser(@Auth AuthUser authUser) {
-        List<PendingCase> pendingCases = pendingCaseService.describeDataRequestPendingCases(authUser);
-        return Response.ok().entity(pendingCases).build();
-    }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DataRequestCasesResource.java]: This is still in use here: https://github.com/DataBiosphere/duos-ui/blob/develop/src/libs/ajax.js#L704

**Review by rushtong — CHANGES_REQUESTED** [review]
Lots of good work here, but this PR is too big and requires a significant amount of manual testing to ensure no regressions. We should break this up into separate PRs, one per endpoint that we're removing.

File: src/main/java/org/broadinstitute/consent/http/service/ElectionService.java [hunk 4]
```diff
@@ -510,8 +508,8 @@ private void setGeneralFields(Election election, String referenceId, ElectionTyp
                 break;
             case DATA_ACCESS:
             case RP:
-                Document dar = dataAccessRequestService.getDataAccessRequestByReferenceIdAsDocument(referenceId);
-                List<Integer> datasetIdList = DarUtil.getIntegerList(dar, DarConstants.DATASET_ID);
+                DataAccessRequest dar = dataAccessRequestService.findByReferenceId(referenceId);
+                List<Integer> datasetIdList = Objects.nonNull(dar) && Objects.nonNull(dar.getData()) ? dar.getData().getDatasetIds() : Collections.emptyList();
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/ElectionService.java:508]: Just a thought - we do this null check quite often. Not for this PR, but a nice refactoring would be to have a `getDatasetIds()` method on DataAccessRequest that would do all the checking for us. That in itself is a pretty big change across lots of classes.

File: src/main/java/org/broadinstitute/consent/http/service/EmailNotifierService.java [hunk 5]
```diff
@@ -217,17 +207,6 @@ public void sendAdminFlaggedDarApproved(String darCode, List<User> admins, Map<U
         }
     }
 
-    public void sendUserDelegateResponsibilitiesMessage(User user, Integer oldUser, String newRole, List<Vote> delegatedVotes) throws MessagingException, IOException, TemplateException {
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/EmailNotifierService.java:220]: Nice ... this is such an outdated method - we removed delegation at least a year ago.

File: src/main/java/org/broadinstitute/consent/http/service/SummaryService.java [hunk 6]
```diff
@@ -37,9 +38,6 @@
 import org.broadinstitute.consent.http.models.Summary;
 import org.broadinstitute.consent.http.models.User;
 import org.broadinstitute.consent.http.models.Vote;
-import org.broadinstitute.consent.http.util.DarConstants;
-import org.broadinstitute.consent.http.util.DarUtil;
-import org.bson.Document;
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/SummaryService.java:42]: 😄

File: src/main/java/org/broadinstitute/consent/http/service/SummaryService.java [hunk 7]
```diff
@@ -295,23 +293,24 @@ public File describeDataAccessRequestSummaryDetail() {
                                     summaryWriter.write("-" + SEPARATOR);
                                     summaryWriter.write("-" + SEPARATOR);
                                 }
-                                summaryWriter.write( dar.get(DarConstants.INVESTIGATOR)  + SEPARATOR);
-                                summaryWriter.write( dar.get(DarConstants.PROJECT_TITLE)  + SEPARATOR);
-                                List<Integer> dataSetIds = DarUtil.getIntegerList(dar, DarConstants.DATASET_ID);
+                                User u = userDAO.findUserById(dar.getUserId());
+                                summaryWriter.write( u.getDisplayName()  + SEPARATOR);
+                                summaryWriter.write( dar.getData().getProjectTitle() + SEPARATOR);
+                                List<Integer> dataSetIds =  dar.getData().getDatasetIds();
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/SummaryService.java]: nit, this isn't necessary since you have `List<Integer> datasetId = dar.getData().getDatasetIds();` on line 273. The original code was making the same mistake.

File: src/main/java/org/broadinstitute/consent/http/service/SummaryService.java [hunk 8]
```diff
@@ -510,11 +509,11 @@ private void setSummaryHeaderDataAccessRequest(FileWriter summaryWriter , Intege
...
-    private Document findAssociatedDAR(List<Document> dataAccessRequests, String referenceId) {
-        Optional<Document> documentOption = dataAccessRequests.stream().
-                filter(d -> d.getString(DarConstants.REFERENCE_ID).equalsIgnoreCase(referenceId)).
+    private DataAccessRequest findAssociatedDAR(List<DataAccessRequest> dataAccessRequests, String referenceId) {
+        Optional<DataAccessRequest> darOption = dataAccessRequests.stream().
+                filter(d -> Objects.nonNull(d) ? d.getReferenceId().equalsIgnoreCase(referenceId) : false).
                 findFirst();
-        return documentOption.orElse(null);
+        return darOption.orElse(null);
     }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/SummaryService.java:516]: This method is another opportunity for refactoring. Instead of calling this method, the caller should instead create a `Map<String, DataAccessRequest>` and just access the map by the reference id. Not for this PR unless you're up for it.

**Review by rushtong — APPROVED** [review]
Nice work!
